### DISCUSSION TITLE
Remove need to set a default value in a switch expression when all possible solutions are provided

### DISCUSSION THREAD

**Discussion by royberris** [dotnet/csharplang#4875] [General] [answered]
I like the (new) switch expressions, but I don't understand why it wants me to implement a default when all possible solutions are provided when working with an enum.

```c#
public enum MyEnum
{
  A,
  B,
  C
}
```

```c#
MyEnum myEnum= MyEnum.A;

string enumName = myEnum switch
{
  MyEnum.A => "a",
  MyEnum.B => "b",
  MyEnum.C => "c",
  _ => "why is this warning here for a default that is met"
}
```

I don't understand why this is the way it is, that is why I put this in general and not in feature request. Also why can we not use `A`, `B`, `C` directly in the scope of the switch expression and we still need to specify the enum type.

**Comment by ZacharyPatten** [comment, marked as answer]
Even though you only have the values `A`, `B`, and `C` defined, the value of any `MyEnum` instance could be an undefined value. This is why the `Enum.IsDefined` method exists.
```cs
using System;

MyEnum myenum = (MyEnum)(-1);
Console.WriteLine(myenum);
Console.WriteLine(myenum is MyEnum.A || myenum is MyEnum.B || myenum is MyEnum.C);
Console.WriteLine(Enum.IsDefined<MyEnum>(myenum));
Console.WriteLine(myenum switch
	{
		MyEnum.A => "a",
		MyEnum.B => "b",
		MyEnum.C => "c",
		_ => "undefined"
	});

public enum MyEnum
{
	A,
	B,
	C
}
```
`Enum.IsDefined`: <https://docs.microsoft.com/en-us/dotnet/api/system.enum.isdefined>

**Reply by royberris** [reply]
Didn't know this, thanks. I never work with enums that are "out of range". We use them to store set data in the database using an enum.

**Comment by ufcpp** [comment]
https://github.com/dotnet/csharplang/issues/3179
